pkg-manager/resolve-dependencies/src/index.ts

```diff
@@ -281,6 +280,20 @@ export async function resolveDependencies (
     }
   }))
 
+  let updatedCatalogs: Record<string, Record<string, string>> | undefined
```

Comment:
For how this is used, see:

https://github.com/pnpm/pnpm/pull/9517/files#diff-f07609f178e1cb78da53c4532059db3919c224b798e1be919f703d971e2fd998